### ISSUE TITLE
How to fix segmentation fault when i'm use getContainer

### ISSUE THREAD

**Issue by Dm3Ch** [krakjoe/pthreads#629]
I'm using Symfony 3 and php pthreads in my project. I need to access entity manager throw Threads. I read some way how to do this on StackOverflow and issues tracker of pthreads.

```
class BaseBackgroundThread extends \Thread
{
    private $container;
    private $env;
    private $debug;

    public function __construct($env, $debug)
    {
        $this->env = $env;
        $this->debug = $debug;
    }

    public function run()
    {
        require 'app/autoload.php';
        include_once 'var/bootstrap.php.cache';

        $kernel = new \AppKernel($this->env, $this->debug);
        $kernel->loadClassCache();
        $kernel->boot();

        $this->container = serialize($kernel->getContainer());
    }

    public function start(int $options = null)
    {
        return parent::start(PTHREADS_INHERIT_NONE);
    }
}
```

But there is one problem. I've got `segmentation fault (core dumped)` error when i'm calling `$kernel->getContainer()`

PHP Version: `PHP 7.0.5-4+donate.sury.org~xenial+1 (cli) ( ZTS )
`
**Update:**

I realized that $kernel->getContainer() doesn't cause crash. But $this->container = $kernel->getContainer() causes crash.

**Update 2:**
$this->container = $kernel->getContainer() causes crash 
self::$container = $kernel->getContainer() causes crash too 
$container = $kernel->getContainer() doesn't cause crash.

**Update 3:**

I tried to run it on php version PHP 5.6.27RC1 (cli) (built: Oct 12 2016 02:24:27)

And got this error.

`Fatal error: Uncaught exception 'PDOException' with message 'You cannot serialize or unserialize PDO instances' in /root/TheBuggestBot/src/TheBuggestBot/CoreBundle/BackgroundThread/BaseBackgroundThread.php:27 zend_mm_heap corrupted`

**Comment by adam-rocska** [comment]
@Dm3Ch  Version, or anything?

**Comment by Dm3Ch** [comment]
@adam-rocska I updated issue message.

**Comment by krakjoe** [comment]
I won't guess what the problem is.

In order to help, I need reproducing code, not a snippet from some framework, or a sentence describing the problem, but complete, executable, reproducing code: That is the only way I can look at the problem.

I'm not familiar enough with any framework to understand the jargon they are surrounded in, and nor do I want to be.

Please, provide a complete reproducing test case.

**Comment by Dm3Ch** [comment]
https://github.com/Dm3Ch/TheBuggestBot/tree/wastelandGames-dev

Run command:
`php bin/console ircbot:run dm3ch -i`

**Comment by Dm3Ch** [comment]
So, have anyone any ideas how can I fix this problem?

**Comment by krakjoe** [comment]
As far as I can tell, this was fixed when we required 7.2 ... if I'm wrong, please re-open the issue, with minimal reproducing code attached ...
